### ISSUE TITLE
Review CodeQL configuration for potential updates

### ISSUE THREAD

**Issue by zhongruige** [woothemes/woocommerce#42403]
We should look into he CodeQL configuration to check for any potential updates that we can make. CodeQL includes both TypeScript and React, which we can also investigate along with this review.

**Comment by Aljullu** [comment]
Hey there, @zhongruige! I might be missing some context here, but are you expecting us to do some work related to this? It's not clear to me from the issue description.

**Comment by tarunvijwani** [comment]
@zhongruige Thank you for logging the issue. I'm moving this task to the cooldown. 

@Aljullu, here is the Slack conversation about it: p1681981771619739-slack-C8X6Q7XQU

**Comment by Aljullu** [comment]
Thanks! :slightly_smiling_face:

**Comment by wavvves** [comment]
I added a comment on the thread p1682083643614709/1681981771.619739-slack-C8X6Q7XQU

@zhongruige can you share the resources on CodeQL and React? 🤔

**Comment by zhongruige** [comment]
Howdy @wavvves!

Here's some resources for this:
- https://codeql.github.com/docs/
- JavaScript specifically: https://codeql.github.com/codeql-query-help/javascript/

I think React is included with the CodeQL JavaScript library so if we have that language enabled in the workflow that should be enough: https://codeql.github.com/codeql-standard-libraries/javascript/index.html

Let me know if this helps or if I can provide any more details.

**Comment by nerrad** [comment]
Just to clarify, Greg created this issue at my request (while I was AFK) because I was pinged about the usage of CodeQL and whether it'd be useful for Woo Blocks (which I was happy to note is _already in use here_). While quickly reviewing our implementation, I noticed I had added it about 2 years ago and thought it'd be worthwhile reviewing to ensure our usage is still valid and current (especially since TypeScript is something I didn't see explicitly included in the config).

I don't anticipate this should take too long (might not actually need anything done).
